sui-execution/latest/sui-move-natives/src/test_scenario.rs

```diff
@@ -502,6 +553,149 @@ pub fn was_taken_shared(
     ))
 }
 
+// native fun was_allocated_receiving_ticket(id: ID): bool;
+pub fn was_allocated_receiving_ticket(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert!(ty_args.is_empty());
+    let id = pop_id(&mut args)?;
+    assert!(args.is_empty());
+    let object_runtime: &ObjectRuntime = context.extensions().get();
+    let was_allocated = object_runtime
+        .test_inventories
+        .allocated_tickets
+        .contains_key(&id);
+    Ok(NativeResult::ok(
+        legacy_test_cost(),
+        smallvec![Value::bool(was_allocated)],
+    ))
+}
+
+pub fn allocate_receiving_ticket_for_object(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    let ty = get_specified_ty(ty_args);
+    let id = pop_id(&mut args)?;
+
+    let Some((tag, layout, _)) = get_tag_and_layouts(context, &ty)? else {
+        return Ok(NativeResult::err(
+            context.gas_used(),
+            E_UNABLE_TO_ALLOCATE_RECEIVING_TICKET,
```

Comment:
Given the native functions return these error codes, I'm now a little confused why the Move code returns error codes as well -- do we need both?

Reply:
Nope! Will remove `was_allocated_receiving_ticket` fn :)